sentinel/src/main/kotlin/com/infinum/sentinel/data/models/memory/triggers/shared/SensorTrigger.kt

```diff
@@ -20,11 +20,16 @@ internal abstract class SensorTrigger(
 
     abstract fun processEvent(event: SensorEvent): Boolean
 
+    private var activeHolder: Boolean? = null
+
     override fun start() {
         sensorManager = (context.getSystemService(Context.SENSOR_SERVICE) as? SensorManager)
         sensorManager?.let {
             registerSensor(it)
-            this.active = true
+            if (activeHolder == null) {
+                activeHolder = active
+            }
+            this.active = activeHolder ?: true // use remembered active state
```

Comment:
So this is problematic part of the code:
```
override fun start() {
           ...
            this.active = true
}
```
This is result with code with reverted [SensorTrigger.kt](https://github.com/infinum/android-sentinel/pull/48/files/3d28a7efeb7ded3c82ee356646e3d84ed6228490#diff-07b4f2abb8a31735103e543ec38cec4b760f343352df1261b423295ca382a07e) changes.
Imagine I am activating proximity sensor after I am bringing app from background
https://github.com/infinum/android-sentinel/assets/20267231/cdc7e47a-e23a-486c-90c7-ae1ec3b549dc

Reply:
My suggestion was to completely remove the flipping of the active flag. So start and stop method would look something like this. And the flag setting would be removed from other methods as well 🤷‍♂️ 

```
    override fun start() {
        sensorManager = (context.getSystemService(Context.SENSOR_SERVICE) as? SensorManager)
        sensorManager?.let {
            registerSensor(it)
        }
    }

    override fun stop() {
        queue?.clear()
        unregisterSensor()
        sensorManager = null
    }
```

Reply:
I think I tried this, but I found out that flipping of flag is necessary

Reply:
Please explain why :D

Reply:
Well, there were some bugs in initialization. I think I saw it. I will try to replicate the issues. 
Also, file history also suggests at bugs with initialization.
https://github.com/infinum/android-sentinel/commits/3d28a7efeb7ded3c82ee356646e3d84ed6228490/sentinel/src/main/kotlin/com/infinum/sentinel/data/models/memory/triggers/shared/SensorTrigger.kt

Reply:
Well, now I can't replicate it 😅

Reply:
I guess we can remove active flipping and see if anyone reports issue 🤷‍♂️

Reply:
registerSensor also has the flipping to false. I'd remove every occurrence of writing to that flag from this class, otherwise the flag would be flipped incorrectly again in some cases.

Reply:
Well I wouldn't change that because it is fallback if sensor manager can't find that sensor

Reply:
Okay, that case is not that important, but what would be the downside of removing also that? The trigger would appear to be active but it wouldn't report anything as the sensor is not working.

Reply:
Maybe only it would be confusing to the user in Sentinel UI?
Not sure